lib/install/deps.js

```diff
@@ -74,7 +74,7 @@ function doesChildVersionMatch (child, requested, requestor) {
     return false
   }
   try {
-    return semver.satisfies(child.package.version, requested.fetchSpec)
+    return semver.satisfies(child.package.version, requested.fetchSpec, true)
```

Comment:
The `loose` option does not seem to be [documented on node-semver](https://github.com/npm/node-semver), so this change would be relying on undocumented behavior.

Also, you mentioned this change allows people to put _garbage _ in their `package.json` files, which doesn't seem like a good investment in terms of what code should go into `npm`.

Lastly, this change does not have any unit tests to prevent a future regression.